packages/eslint-plugin-template/tests/rules/no-call-expression/cases.ts

```diff
@@ -228,4 +228,61 @@ export const invalid: readonly InvalidTestCase<MessageIds, Options>[] = [
       { char: '*', messageId },
     ],
   }),
+
+  convertAnnotatedSourceToFailureCase({
+    description: 'should fail for calls in arguments',
+    annotatedSource: `
+    {{ foo(bar(), baz()) }}
+       ~~~~^^^^^~~#####~
+    `,
+    messages: [
+      { char: '~', messageId },
+      { char: '^', messageId },
+      { char: '#', messageId },
+    ],
+  }),
+  extendError(
+    convertAnnotatedSourceToFailureCase({
+      description: 'should fail multiple times for nested calls',
+      annotatedSource: `
+    {{ foo()() }}
+       ^^^^^~~
+    `,
+      messages: [
+        { char: '~', messageId },
+        { char: '^', messageId },
+      ],
+    }),
+  ),
+  extendError(
+    convertAnnotatedSourceToFailureCase({
+      description: 'should fail multiple times for chained calls',
+      annotatedSource: `
+    {{ foo().bar() }}
+       ^^^^^~~~~~~
+    `,
+      messages: [
+        { char: '~', messageId },
+        { char: '^', messageId },
+      ],
+    }),
+  ),
+  convertAnnotatedSourceToFailureCase({
+    description: 'should fail for nested call not in allowList',
+    annotatedSource: `
+    {{ notOk().ok() }}
+       ~~~~~~~
+    `,
+    options: [{ allowList: ['ok'] }],
+    messages: [{ char: '~', messageId }],
+  }),
 ];
+
+// convertAnnotatedSourceToFailureCase unfortunately cannot handle multiple errors starting/ending at the same position;
```

Comment:
It's our own utility, so feels a little weird to add a comment saying "unfortunately cannot handle". Can we make it handle it? 🤔

Reply:
I looked at the implementation, but only briefly, I didn't really want this PR to turn into a side-quest of re-writing a test utility to handle just these few cases. 

If you feel that it's worthwhile, I can try and implement the necessary changes.

Reply:
It's a unit test utility right, so a super fast feedback loop - if the existing tests and the new tests pass then the patch is "correct" :)

If it bogs you down for a long time then let me know, but it's definitely worth giving it a go

Reply:
Alright, I think I've managed to finally figure it out - this hack is no longer necessary and the other tests didn't break.

The code for processing annotated cases assumed the code would always be at `line-1`, but this is not true if we have multiple lines of annotations, I changed it to instead keep track of the last non-annotation line and use that for setting start and end line positions